lib/utils.pm

```diff
@@ -831,39 +834,128 @@ sub minimal_patch_system {
     }
 }
 
-=head2 zypper_search
+=head2 parse_zypper_table
 
- zypper_search($search_params);
+ parse_zypper_table($table, $fields);
 
-Run zypper search with given command line arguments and parse the output into
-an array of hashes.
+Parse ASCII table generated by commands such as C<zypper search> into an array
+of hashes. Pass the ASCII table in C<$table> as a single string. Pass an array
+of hash keys in C<$fields>. Keys will be mapped to columns by position.
 
 =cut
 
-sub zypper_search {
-    my $params = shift;
-    my @fields = ('status', 'name', 'type', 'version', 'arch', 'repository');
+sub parse_zypper_table {
+    (my $table, my $fields) = @_;
     my @ret;
 
-    my $output = script_output("zypper -n se $params");
-
-    for my $line (split /\n/, $output) {
+    for my $line (split /\n/, $table) {
+        $line =~ s/^\s*(.*?)\s*$/$1/;
         my @tokens = split /\s*\|\s*/, $line;
-        next if $#tokens < $#fields;
+        next if $#tokens < $#$fields;
         my %tmp;
 
-        for (my $i = 0; $i < scalar @fields; $i++) {
-            $tmp{$fields[$i]} = $tokens[$i];
+        for (my $i = 0; $i < scalar @$fields; $i++) {
+            $tmp{$$fields[$i]} = $tokens[$i];
         }
 
         push @ret, \%tmp;
     }
 
-    # Remove header from package list
+    # Remove header from row list
     shift @ret;
     return \@ret;
 }
 
+=head2 zypper_search
+
+ zypper_search($search_params);
+
+Run C<zypper search> with given command line arguments and parse the output
+into an array of hashes.
+
+=cut
+
+sub zypper_search {
+    my $params = shift;
+    my %opts;
+    my @fields = ('status', 'name', 'summary', 'type');
+
+    # Set Getopt to ignore any unrecognized options
+    Getopt::Long::Configure('bundling', 'pass_through', 'permute');
+
+    # Call in array context to silence warnings about extra options and args
+    my @tmp = GetOptionsFromString($params, \%opts, 'details|s', 'verbose|v');
```

Comment:
`@tmp` is not used, right? Do you really need the assignment?

Reply:
As the comment above says, I need to call `GetOptionsFromString()` in array context, otherwise it'll print warnings about unknown options and leftover arguments into log. Do you have any better idea than assigning to an unused variable?

Reply:
i read that. but i dont think you need the assignment. Sorry if it is kinda picky, i just thought to bring it up. but no problem if you want to keep it as it is

Reply:
Well, if there is no explicit switch to array context, the function will be called in scalar context and print warnings. I need either the assignment or something else that forces array context. Do you have any suggestion what that "something else" could be?